src/sentry/deletions/defaults/group.py

```diff
@@ -256,10 +258,38 @@ def delete_group_hashes(
             logger.warning("Error scheduling task to delete hashes from seer")
         finally:
             hash_ids = [gh[0] for gh in hashes_chunk]
+            if options.get("deletions.group.delete_group_hashes_metadata_first"):
+                # If we delete the grouphash metadata rows first we will not need to update the references to the other grouphashes.
+                # If we try to delete the group hashes first, then it will require the updating of the columns first.
+                #
+                # To understand this, let's say we have the following relationships:
+                # gh A -> ghm A -> no reference to another grouphash
+                # gh B -> ghm B -> gh C
+                # gh C -> ghm C -> gh A
+                #
+                # Deleting group hashes A, B & C (since they all point to the same group) will require:
```

Comment:
This function is called as this: `delete_group_hashes(project_id, group_ids)`, thus, all group hashes point to the same group.

<img width="599" height="58" alt="Image" src="https://github.com/user-attachments/assets/47a02914-6755-43d4-a931-ca9c7d623699" />